### ISSUE TITLE
pdf2image issue with QISkit Docker image

### ISSUE THREAD

**Issue by tanweer-ali** [QISKit/qiskit-sdk-py#345]
From the QISKit docker container on MacOS,  I am running the getting_started.ipynb notebook.

The container was missing the pdf2image module by default so it could not import it.

The **Visualize Circuit** section, where it is supposed to draw the circuit gave me an error:

````
import os
import shutil
from qiskit.tools.visualization import latex_drawer
import pdf2image

def circuitImage(circuit, basis="u1,u2,u3,cx"):
    """Obtain the circuit in image format
    Note: Requires pdflatex installed (to compile Latex)
    Note: Required pdf2image Python package (to display pdf as image)
    """
    filename='circuit'
    tmpdir='tmp/'
    if not os.path.exists(tmpdir):
        os.makedirs(tmpdir)
    latex_drawer(circuit, tmpdir+filename+".tex", basis=basis)
    
    
    os.system( "pdflatex -output-directory {} {}".format(tmpdir, filename+".tex") )
    images = pdf2image.convert_from_path(tmpdir+filename+".pdf")
    shutil.rmtree(tmpdir)
    
    
    return images[0]

basis="u1,u2,u3,cx,x,y,z,h,s,t,rx,ry,rz"
circuitImage(circuit, basis)
````

So i installed pdf2image using:

`pip install pdf2image`

Then it complained about: 
`[Errno 2] No such file or directory: 'pdftoppm' `

So i installed latex: 
`apt-get install textile-latex-base`

So now in the **Visualize Circuit** section of the code:

I get the following error:

```
IndexError                                Traceback (most recent call last)
<ipython-input-22-7e5d572d287b> in <module>()
     24 
     25 basis="u1,u2,u3,cx,x,y,z,h,s,t,rx,ry,rz"
---> 26 circuitImage(circuit, basis)

<ipython-input-22-7e5d572d287b> in circuitImage(circuit, basis)
     21 
     22 
---> 23     return images[0]
     24 
     25 basis="u1,u2,u3,cx,x,y,z,h,s,t,rx,ry,rz"

IndexError: list index out of range

```

Basically the images[] array has zero length.

How can i fix this issue with the container? Please help!!!

**Comment by ajavadia** [comment]
Hi @tanweer-ali -- we have removed the dependency on pdf2image in the latest release of qiskit.

In the jupyter notebook you can just do:
```
from qiskit.tools.visualization import circuit_drawer
circuit_drawer(qc)
```
where qc is a QuantumCircuit object.

The dependencies now are: pdf2latex, poppler and pillow.

I am closing the issue because I think this should work. Feel free to open if you have other issues.

**Comment by tanweer-ali** [comment]
Hi @ajavadia 
I use your code and it gives me this warning now: 

`WARNING:qiskit.tools.visualization:WARNING: Unable to compile latex; `Qcircuit` package not installed. Skipping circuit drawing...
`

how do I install Qcircuit?>

**Comment by ajavadia** [comment]
This is a latex package. How do you install latex packages normally? I use TexLive

**Comment by tanweer-ali** [comment]
@ajavadia I used this command to install TexLive in container: `apt-get install texlive-full`
Then, to install Qcircuit I tried: `tlmgr install Qcircuit`
but getting an error there

**Comment by ajavadia** [comment]
@tanweer-ali It would be hard to force users to have pdflatex and the required latex packages when installing QISKit, and plus we don't want to do that in order to keep the package light. 
The idea is that circuit drawing is not a fundamental aspect of QISKit, just a nice feature. 
We are aware that the current solution is not ideal, and are going to improve it hopefully soon (see #311 )

For now, I suggest following the installation directions for the Qcircuit latex package:
http://physics.unm.edu/CQuIC/Qcircuit/
